### DISCUSSION TITLE
move away from make

### DISCUSSION THREAD

**Discussion by devkral** [dymmond/esmerald#306] [Ideas]
I think it would be better for crossplattform support to move away from make.
Windows users will swear if they have to install another project.

Maybe there is something more pythonic, the best thing would be if hatch would support some added commands out of the box (like npm). I am not sure if hatch supports such a feature.

**Comment by tarsil** [comment]
I understand. We added some windows support scripts recently because of that but regarding hatch I'm not 100% sure but nothing like searching and see better alternatives.

The make is more universal amongst Mac and Linux users.

**Reply by devkral** [reply]
actually hatch can do that and per environment:
https://hatch.pypa.io/latest/environment/

**Reply by devkral** [reply]
why don't you use python scripts instead of ps/shell scripts in your scripts directory?

Anyway: I am on replacing the scripts with hatch here as well

**Reply by tarsil** [reply]
@devkral understood but there are historical reasons for it. 

https://github.com/github/scripts-to-rule-them-all

This was what we were basing on. Why don't you join us in your discord? It would be great to have you there and we could discuss a lot of subjects directly with the community.

**Reply by devkral** [reply]
Sound good, but I didn't saw an invitation link.

Note: I prefer matrix, because of the freedom (you heard about the dudes talking about development kicked of discord by nintendo) and better integration

**Reply by tarsil** [reply]
Got it and I'm aware but Esmerald discord was created in 2022 and for now, discord is serving our purposes.

The link was in the sponsorship section of the documentation but now here 👍🏼

**Reply by devkral** [reply]
I skimmed through the hatch documentation, there are some cool features:

- self-referencing in deps (you can use the extras in the test env e.g. by specifying esmerald[all])
- linting with ruff (hatch fmt), no need for extra deps
- (since a few days ago) `hatch test`: tests with pytest, you can customize the env by specifying a special one named hatch-test
- (since a few days ago): `hatch run` can now auto install deps defined in scripts (like pipx, I think the syntax is the same). This is the make killer. Why not using plain python scripts for installing and co? No need anymore to care for newlines

**Reply by tarsil** [reply]
That is a good idea @devkral but we also need to incrementally do it since some people also have it and we need to have some extensive documentation in the contribution to make sure people can do it properly. Not everyone is familiar with hatch and 99% of the developers still use the classic venv manually.

**Comment by tarsil** [comment]
This is done 👍🏼
